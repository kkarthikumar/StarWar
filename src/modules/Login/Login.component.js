import React, { useState, memo, useContext } from "react";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import history from "../../history";
import Grid from "@material-ui/core/Grid";
import { withStyles, Box } from "@material-ui/core";
import styles from "./login.style";
import isEmpty from "lodash/isEmpty";

const LoginComponent = props => {
  const [state, setState] = useState({ userName: "", password: "" });
  const [errorMessage, setErrorMessage] = useState(false);

  const { isLoggedInError, onValidateUser, isLoggedIn, classes } = props;
  const { userName, password } = state;
  const handleChange = (value, fieldName) => {
    setState({ ...state, [fieldName]: value });
  };

  const onSubmit = async () => {
    if (!userName || !password) {
      setErrorMessage("username and password is requried");
      return;
    }

    await onValidateUser(state);
  };

  return (
    <>
      <Grid container className={classes.parent}>
        <Grid item xs={12} md={7}>
          <Typography variant="h4">Login</Typography>
        </Grid>
        <Grid item xs={12} md={7}>
          <TextField
            id="user-name"
            label="username"
            className={classes.textField}
            error={!state.userName && errorMessage}
            value={userName}
            onChange={e => handleChange(e.target.value, "userName")}
            margin="normal"
          />
        </Grid>
        <Grid item xs={12} md={7}>
          <TextField
            type="password"
            id="password"
            label="password"
            className={classes.textField}
            error={!state.password && errorMessage}
            value={password}
            onChange={e => handleChange(e.target.value, "password")}
            margin="normal"
          />
        </Grid>
        <Grid item xs={12} md={7}>
          <Typography variant="caption" style={{ color: "red" }}>
            {isLoggedInError
              ? "username or password is incorrect"
              : errorMessage}
          </Typography>
        </Grid>
        {isLoggedIn && history.push("/search")}
        <Grid item xs={12} md={7}>
          <Box pt={2}>
            <Button
              disabled={isEmpty(state.userName) || isEmpty(state.password)}
              className={classes.button}
              variant="contained"
              color="primary"
              onClick={onSubmit}
            >
              Submit
            </Button>
          </Box>
        </Grid>
      </Grid>
    </>
  );
};

export default memo(withStyles(styles)(LoginComponent));
