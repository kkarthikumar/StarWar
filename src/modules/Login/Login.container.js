import React, { memo, useState } from "react";

import LoginComponent from "./Login.component";
import * as LoginActionCreators from "./../../store/actions/login.action";

import { bindActionCreators } from "redux";
import PropTypes from "prop-types";

import { connect } from "react-redux";
import { compose } from "redux";

// import * as actions from '..';

const LoginContainer = props => {
  const {
    isLoggedInError,
    actions: { onValidateUser },
    isLoggedIn,
    history
  } = props;
  return (
    <LoginComponent
      isLoggedIn={isLoggedIn}
      onValidateUser={onValidateUser}
      isLoggedInError={isLoggedInError}
      history={history}
    />
  );
};

const mapStateToProps = state => {
  return {
    isLoggedInError: state.app.isLoggedInError,
    isLoggedIn: state.app.isLoggedIn
  };
};

const mapDispatchToProps = dispatch => {
  return {
    actions: { ...bindActionCreators(LoginActionCreators, dispatch) }
  };
};

LoginContainer.propTypes = {
  actions: PropTypes.object.isRequired,
  classes: PropTypes.object,
  isLoggedInError: PropTypes.bool.isRequired,
  isLoggedIn: PropTypes.bool.isRequired,
  history: PropTypes.object.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(memo(LoginContainer));
