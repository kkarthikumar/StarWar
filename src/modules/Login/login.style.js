export const styles = theme => ({
  textField: {
    width: "100%"
  },
  button: {
    float: "none",
    position: "static",
    display: "block",
    margin: "auto",
    width: "max-content",
    marginBottom: "32"
  },
  parent: {
    display: "flex",
    justifyContent: "center",
    marginTop: "200px"
  }
});

export default styles;
