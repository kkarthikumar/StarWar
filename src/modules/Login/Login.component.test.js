import React from "react";
import Enzyme, { mount } from "enzyme";
import Login from "./Login.component";

import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("<Login /> with no props", () => {
  const container = mount(<Login />);
  console.log(container);
  it("should have a password field", () => {
    /* Similar to above */
    expect(container.find('input[type="password"]').length).toEqual(1);
  });

  it("should have a username text field", () => {
    /* */
    expect(container.find('input[type="text"]').length).toEqual(1);
  });
});
