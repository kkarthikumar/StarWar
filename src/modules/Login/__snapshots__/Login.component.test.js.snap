// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> with no props should match the snapshot 1`] = `"<div class=\\"MuiGrid-root LoginComponent-parent-3 MuiGrid-container\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7\\"><h4 class=\\"MuiTypography-root MuiTypography-h4\\">Login</h4></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7\\"><div class=\\"MuiFormControl-root MuiTextField-root LoginComponent-textField-1 MuiFormControl-marginNormal\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" for=\\"user-name\\">username</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input\\" id=\\"user-name\\" value=\\"\\"/></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7\\"><div class=\\"MuiFormControl-root MuiTextField-root LoginComponent-textField-1 MuiFormControl-marginNormal\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" for=\\"password\\">password</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input\\" id=\\"password\\" value=\\"\\"/></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7\\"><span class=\\"MuiTypography-root MuiTypography-caption\\" style=\\"color:red\\"></span></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-align-content-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-md-7\\"><div class=\\"MuiBox-root MuiBox-root-194\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary LoginComponent-button-2 Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiButton-label\\">Submit</span></button></div></div></div>"`;
